rwmutex/rwmutex_test.go

```diff
@@ -131,3 +131,22 @@ func TestRWMutex(t *testing.T) {
 	HammerRWMutex(10, 10, n)
 	HammerRWMutex(10, 5, n)
 }
+
+func TestWriteWriteReadDeadlock(t *testing.T) {
+	defer runtime.GOMAXPROCS(runtime.GOMAXPROCS(-1))
```

Comment:
It is possible to simply write this part as:  `defer runtime.GOMAXPROCS(runtime.GOMAXPROCS(2))`